runtime/Makefile

```diff
@@ -35,23 +35,23 @@ all: $(CHPL_ENV_HEADER) third-party-pkgs all.helpme
 depend:
 
 clean: clean.helpme
-	rm -f tags TAGS
+	rm -f tags TAGS $(CHPL_ENV_HEADER)
```

Comment:
I may be not understanding something subtle about how these `make` targets work, but why put `CHPL_ENV_GEN` here explicitly instead of just including it in `CLEAN_TARGS` in `runtime/make/Makefile.runtime.head`, perhaps in place of the now-unused `GEN_HEADERS`.  Then the existing `clean` target in `runtime/make/Makefile.runtime.commonrules` would remove it, wouldn't it?  (And similarly for `tags` and `TAGS` here, I guess.)

Reply:
I don't think we'd want to add it explicitly to CLEAN_TARGS in runtime/make/Makefile.runtime.head because that Makefile is included by all other Makefiles in the runtime, so I believe a potential result is that each sub-Makefille would try to remove the file redundantly?  (not a problem per se, but overkill... and since this is the Makefile that created it, I think it's the one that should be cleaning it up).

I tried adding it to GEN_HEADERS and that didn't seem to work (though I could've made an error).  My guess was that the CLEAN_TARGS variable was only currently used by the subdirectory Makefiles and not this top-level one and that that's why TAGS was special-cased here as well.  But it's been a long time since I've looked at these Makefiles enough to be confident that I've got that right and I wasn't interested in re-engineering them, just closing what seemed to be a silly issue.  I also think the current approach is arguably more obvious/straightforward since what it's doing is obvious in the source code.

Reply:
Ah, interesting.  For some reason I've been laboring under a misconception about how clean/cleanall/clobber work.  None of those `rm` the installed library.  They just remove (via the `Makefile.runtime.commonrules` `clean` target) the object files that would go into creating that library (plus some other stuff).  Then any subsequent make rebuilds the library simply because the objects upon which it depends aren't there.

But `chpl-env-gen.h` is actually functionally rather like the library, not the object files.   (As are `tags`/`TAGS` also.)  Okay, good enough.  Best to leave it as it stands.